hazelcast/src/main/java/com/hazelcast/config/TopicConfig.java

```diff
@@ -113,11 +115,43 @@ public boolean isGlobalOrderingEnabled() {
      * @return The updated TopicConfig
      */
     public TopicConfig setGlobalOrderingEnabled(boolean globalOrderingEnabled) {
-        this.globalOrderingEnabled = globalOrderingEnabled;
+    	if (this.multiThreadingEnabled && globalOrderingEnabled) {
+            throw new IllegalArgumentException("Global ordering can not be enabled when multi-threading is used.");
+    	} 
+    	this.globalOrderingEnabled = globalOrderingEnabled;
         return this;
     }
+    
+    /**
+     * Checks if multi-threaded processing of incoming messages is enabled or not. 
+     * When disabled only one dedicated thread will handle all topic messages. Otherwise
+     * any thread from events thread pool can be used for message handling.
+     *
+     * @return true if multi-threading is enabled, false if disabled
+     */
+    public boolean isMultiThreadingEnabled() {
+        return multiThreadingEnabled;
+    }
+    
 
     /**
+     * Enable multi-threaded message handling. When enabled any thread from events 
+     * thread pool can be used for incoming message processing. Otherwise only one
+     * dedicated thread will be used to handle topic messages.    
+     * Note: it can be enabled only in case when global ordering is disabled.
+     *
```

Comment:
You are only mentioning that the global order will be broken when the multi-threading enabled. We also have an implicit order which is same publisher's messages are consumed in order, which multi-threading will also break this one.